tasks/mynt.ts

```diff
@@ -993,3 +993,107 @@ task(
       logger.warn("====================================");
     }
   });
+
+task("upgrade:mocIntegration", "Upgrade implementation of mocIntegration contract")
+.addOptionalParam("isMultisig", "flag if transaction needs to be intiated from the multisig contract")
+.addOptionalParam("isSIP", "flag if transaction needs to be initiated from the SIP")
+.setAction(async ({ isMultisig, isSIP }, hre) => {
+  const { ethers, deployments: { get }, getNamedAccounts } = hre;
+  const { deployer } = await getNamedAccounts();
+  const myntAdminProxy = await ethers.getContract("MyntAdminProxy");
+  const mocIntegrationProxy = await ethers.getContract("MocIntegration"); // MocIntegration
+
+  const MocIntegrationFactory = await ethers.getContractFactory("MocIntegration");
+  const newMocIntegrationImpl = await MocIntegrationFactory.deploy();
+  console.log(`Upgrading mocIntegration implementation to ${newMocIntegrationImpl.address}`)
+
+  if(isMultisig) {
+    const multisigAddress = (await get("MultisigWallet")).address;
+    const dataUpgrade = myntAdminProxy.interface.encodeFunctionData("upgrade", [
+      mocIntegrationProxy.address, newMocIntegrationImpl.address
+    ]);
+
+    await sendWithMultisig(
+      hre,
+      multisigAddress,
+      myntAdminProxy.address,
+      dataUpgrade,
+      deployer
+    );
+  } else if(isSIP) {
+    const signatureUpgrade = "upgrade(address,address)";
+    const dataUpgrade = myntAdminProxy.interface.encodeFunctionData("upgrade", [
+      mocIntegrationProxy.address, newMocIntegrationImpl.address
+    ]);
+
+    const sipArgs: ISipArgument["args"] = {
+      targets: [mocIntegrationProxy.address],
+      values: [0],
+      signatures: [signatureUpgrade],
+      data: [dataUpgrade],
+      description: "Upgrade MocIntegration contract"
+    }
+
+    logger.warn(">>> CREATE A SIP WITH THIS ARGS: <<<");
+    logger.info(sipArgs);
+    logger.warn("====================================");
+  }
+});
+
+task("mynt:setMassetTokenTransferWithPermit", "set mAsset token transfer with permit address in massetManager contract")
+.addOptionalParam("isMultisig", "flag if transaction needs to be intiated from the multisig contract")
+.addOptionalParam("isSIP", "flag if transaction needs to be initiated from the SIP")
+.setAction(async ({ isMultisig, isSIP }, hre) => {
```

Comment:
pls replace isMultisig and isSIP params with querying owner - if it is TimelockOwner or TimelockAdmin then it is a SIP, if it is multisig, then multisig tx, else just call upgrade function.  
store these key -> deployment in a generic helper - e.g. multisig contract can change, so it should be e.g. {"multisig", "MultiSigWallet"}, {"sip", ["TimelockOwner", "TimelockAdmin"]} and apply this change to all the tasks.

Reply:
updated in this [commit](https://github.com/DistributedCollective/mynt/pull/17/commits/0b495d7b734a64db865d48d07a012ef1cefba6b4)

But, this is only for this `mynt:setMassetTokeentransferWithPermit` task.
I will let u review it first, then only if everything's ok from your side, i will apply to the other task